File: iOS/old/iOS开发指南(从Hello World到App Store上架)/swift/11/11.4.2/MyNotes/NoteBL.m
//
//  NoteBL.m
//  MyNotes
//
//  Created by andyron on 2016/12/20.
//  Copyright © 2016年 andyron. All rights reserved.
//

#import "NoteBL.h"
#import "NoteDAO.h"

@implementation NoteBL
-(NSMutableArray*)createNote:(Note *)model {
    NoteDAO *dao = [NoteDAO sharedManager];
    [dao create:model];
    return [dao findAll];
}

-(NSMutableArray*)remove:(Note *)model {
    NoteDAO *dao = [NoteDAO sharedManager];
    [dao remove:model];
    return [dao findAll];
}

-(NSMutableArray*) findAll {
    NoteDAO *dao = [NoteDAO sharedManager];
    return [dao findAll];
}
@end

File: iOS/old/iOS开发指南(从Hello World到App Store上架)/swift/11/11.4.1/MyNotes/NoteBL.swift
//
//  NoteBL.swift
//  MyNotes
//
//  Created by andyron on 2016/12/19.
//  Copyright © 2016年 andyron. All rights reserved.
//

import Foundation

class NoteBL {
    
    func createNote(model: Note) -> NSMutableArray {
        let dao:NoteDAO = NoteDAO.sharedInstance
        dao.create(model: model)
        return dao.findAll()
    }
    
    func remove(model: Note) -> NSMutableArray {
        let dao:NoteDAO = NoteDAO.sharedInstance
        dao.remove(model: model)
        return dao.findAll()
    }
    
    func findAll() -> NSMutableArray {
        let dao:NoteDAO = NoteDAO.sharedInstance
        return dao.findAll()
    }
}
